app.rb

```diff
@@ -1,9 +1,44 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+require_relative './lib/peeps.rb'
 
 class Chitter < Sinatra::Base
+  enable :sessions
+
+  configure :development do
+    register Sinatra::Reloader
+  end
+
   get '/test' do
     'Test page'
   end
 
+  get '/' do
+    @peeps = Peeps.all
+    erb:index
+  end
+
+  get '/new_peep' do
+    erb:post_a_peep 
+  end
+
+  post '/new_peep' do
+    session[:new_post_message] = params[:new_post_message]
+    Peeps.new_peep(message:session[:new_post_message])
+    redirect ('/') 
+  end
+
+  post '/search' do
+    session[:search] = params[:search]
+    session[:search_results] = Peeps.search(search:session[:search])
```

Comment:
Sessions is one way to do this! Another way, which doesn't keeping track of any state in the server is to use **query** **parameters** (you can do a CTRL-F for that on the Sinatra docs http://sinatrarb.com/intro.html).  A nice bonus is that this allows users to share search result via a link too, which doesn't work with sessions. This is how many search engines handle it too, btw. Here's an example: https://www.google.com/search?q=query+parameters